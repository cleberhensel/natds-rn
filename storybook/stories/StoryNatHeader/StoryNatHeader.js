import React, { useState } from 'react';
import {
  ScrollView,
  SafeAreaView,
  Dimensions,
} from 'react-native';
import { NatContainer, NatHeader } from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatHeader = () => (
  <SafeAreaView style={styles.defaultScreen}>
    <NatContainer>
      <ScrollView>
        <NatHeader
          title="Title Text"
          icon="iconRight"
          onPress={() => alert('press')}
          onPressRight={() => alert('press')}
          textRight="Text Right"
        />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>
);

StoryNatHeader.propTypes = {
};
