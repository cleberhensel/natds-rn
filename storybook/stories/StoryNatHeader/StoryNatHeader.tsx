import React from 'react';
import {
  ScrollView, SafeAreaView, Dimensions, Alert,
} from 'react-native';
import { NatContainer, NatHeader } from '../../../lib';

const styles = {
  defaultScreen: {
    flex: 1,
    width: Dimensions.get('window').width,
  },
};

export const StoryNatHeader: React.SFC<{}> = () => (
  <SafeAreaView style={styles.defaultScreen}>
    <NatContainer>
      <ScrollView>
        <NatHeader
          title="Title Text"
          icon="iconRight"
          onPress={() => Alert.alert('press')}
          onPressRight={() => Alert.alert('press')}
          textRight="Text Right"
        />
      </ScrollView>
    </NatContainer>
  </SafeAreaView>
);
