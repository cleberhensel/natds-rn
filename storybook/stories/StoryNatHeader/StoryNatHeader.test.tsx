import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components/native";
import { StoryNatHeader } from "./StoryNatHeader";
describe("StoryNatHeader component", () => {
  it("Should render StoryNatHeader component ", () => {
    const button = renderer.create(<StoryNatHeader />).toJSON();
    expect(button).toMatchSnapshot("StoryNatHeader snapshot");
  });
});
