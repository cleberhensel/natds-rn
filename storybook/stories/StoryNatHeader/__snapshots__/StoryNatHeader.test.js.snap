// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryNatHeader component Should render StoryNatHeader component  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }
>
  <View
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="small"
    paddingLeft="small"
    paddingRight="small"
    paddingTop="small"
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          accessibilityRole="header"
          accessible={true}
          style={
            Array [
              Object {
                "backgroundColor": "#FFFFFF",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "height": 56,
                "justifyContent": "space-between",
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingTop": 16,
                "width": "100%",
              },
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "flex-start",
                "opacity": 1,
                "width": 24,
              }
            }
          >
            <Image
              accessibilityLabel="navegação para retornar a página anterior"
              accessible={true}
              source={
                Object {
                  "testUri": "../../../lib/assets/icons/nat-ic-right.png",
                }
              }
            />
          </View>
          <Text
            accessibilityLabel="Title Text"
            icon="iconRight"
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "NaturaSans-Regular",
                  "fontSize": 20,
                  "marginLeft": 24,
                },
              ]
            }
          >
            Title Text
          </Text>
          <View
            accessibilityRole="button"
            style={
              Array [
                Object {
                  "color": "#F4AB34",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "flex-end",
                  "width": 60,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "orange",
                    "fontSize": 14,
                  }
                }
              >
                 
                Text Right
                 
              </Text>
              <Image
                accessibilityLabel="realizar ação do item da barra superior direita"
                accessible={true}
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
